src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -236,28 +240,61 @@ public void setIgnoreCommitList(List<CommitHash> ignoreCommitList) {
         return authorList;
     }
 
+    /**
+     * Sets the details of {@code author} to {@code RepoConfiguration} including the default alias, alias
+     * and display name.
+     */
+    public void setAuthorDetails(Author author) {
+        // Set GitHub Id as default alias
+        addAuthorAliases(author, Arrays.asList(author.getGitId()));
+
+        addAuthorAliases(author, author.getAuthorAliases());
+
+        setAuthorDisplayName(author, author.getDisplayName());
+    }
+
+    /**
+     * Propagate the {@code IgnoreGlobList} of {@code RepoConfiguration} to {@code author}.
+     */
+    public void propagateIgnoreGlobList(Author author) {
+        author.appendIgnoreGlobList(this.getIgnoreGlobList());
+    }
+
     public void addAuthor(Author author) {
         authorList.add(author);
+        setAuthorDetails(author);
+        propagateIgnoreGlobList(author);
     }
 
     public boolean containsAuthor(Author author) {
         return authorList.contains(author);
     }
 
+    /**
+     * Clears authors information and sets the {@code authorList} to {@code RepoConfiguration}.
+     */
     public void setAuthorList(List<Author> authorList) {
         this.authorList = authorList;
+        authorAliasMap.clear();
+        authorDisplayNameMap.clear();
 
         authorList.forEach(author -> {
-            // Set GitHub Id as default alias
-            addAuthorAliases(author, Arrays.asList(author.getGitId()));
-
-            setAuthorDisplayName(author, author.getDisplayName());
-
-            // Propagate RepoConfiguration IgnoreGlobList to Author
-            author.appendIgnoreGlobList(this.getIgnoreGlobList());
+            setAuthorDetails(author);
+            propagateIgnoreGlobList(author);
         });
     }
 
+    public void addAuthors(List<Author> authorList) {
```

Comment:
Group `addAuthor()` and `addAuthors()` together by moving `addAuthors()` under `addAuthor()`.